src/NosCore.GameObject/Character.cs

```diff
@@ -645,69 +650,69 @@ public void Buy(Shop shop, short slot, short amount)
             }
 
             short slotChar = item.Slot;
-            List<InventoryItemInstance> inv;
+            List<InventoryItemInstance>? inv;
             if (shop.Session == null)
             {
                 inv = InventoryService.AddItemToPocket(InventoryItemInstance.Create(
-                    ItemProvider.Create(item.ItemInstance.ItemVNum, amount), Session.Character.CharacterId));
+                    ItemProvider.Create(item.ItemInstance!.ItemVNum, amount), Session.Character.CharacterId));
             }
             else
             {
                 if (price + shop.Session.Character.Gold > shop.Session.WorldConfiguration.MaxGoldAmount)
                 {
-                    SendPacket(new SMemoPacket
+                    await SendPacket(new SMemoPacket
                     {
                         Type = SMemoType.FatalError,
                         Message = Language.Instance.GetMessageFromKey(LanguageKey.TOO_RICH_SELLER, Account.Language)
                     });
                     return;
                 }
 
-                if (amount == item.ItemInstance.Amount)
+                if (amount == item.ItemInstance?.Amount)
                 {
                     inv = InventoryService.AddItemToPocket(InventoryItemInstance.Create(item.ItemInstance,
                         Session.Character.CharacterId));
                 }
                 else
                 {
                     inv = InventoryService.AddItemToPocket(InventoryItemInstance.Create(
-                        ItemProvider.Create(item.ItemInstance.ItemVNum, amount), Session.Character.CharacterId));
+                        ItemProvider.Create(item.ItemInstance?.ItemVNum ?? 0, amount), Session.Character.CharacterId));
                 }
             }
 
             if (inv?.Count > 0)
             {
                 inv.ForEach(it => it.CharacterId = CharacterId);
-                var packet = shop.Session?.Character.BuyFrom(item, amount, slotChar);
+                var packet = await (shop.Session == null ? Task.FromResult((NInvPacket?)null) : shop.Session.Character.BuyFrom(item, amount, slotChar));
                 if (packet != null)
                 {
-                    SendPacket(packet);
+                    await SendPacket(packet);
                 }
 
-                SendPackets(
+                await SendPackets(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)